actions/views/channel.js

```diff
@@ -94,12 +95,19 @@ export function leaveChannel(channelId) {
     return async (dispatch, getState) => {
         const state = getState();
         const myPreferences = getMyPreferences(state);
+        const currentUserId = getCurrentUserId(state);
+        const currentTeamId = getCurrentTeamId(state);
 
         if (isFavoriteChannel(myPreferences, channelId)) {
             dispatch(unfavoriteChannel(channelId));
         }
 
         const teamUrl = getCurrentRelativeTeamUrl(state);
+        LocalStorageStore.removePreviousChannelName(currentUserId, currentTeamId);
+        const channel = getChannel(state, channelId);
+        if (LocalStorageStore.getPreviousChannelName(currentUserId, currentTeamId) === channel.name) {
+            LocalStorageStore.removePreviousChannelName(currentUserId, currentTeamId);
```

Comment:
I'm a little confused, what's the purpose of calling this again conditionally if we just called the exact same thing three lines before?

Reply:
You can have the last channel and the penultimate channel, if you are in the channel and you refresh the browser you can end up with the same channel as last and penultimate

Reply:
Maybe we should change it so that you can't end up with the same channel for both of the last and penultimate channels? It might be a bit easier to do stop the app from getting in that state in the first place instead of removing both channels here.

Reply:
Yes, that makes sense, I'll take a look.

Reply:
Done :)